src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.java

```diff
@@ -364,11 +364,23 @@
    
    // Tools
    public abstract AppCommand showShellDialog();
-   public abstract AppCommand newTerminal();
    public abstract AppCommand macPreferences();
    public abstract AppCommand showOptions();
    public abstract AppCommand modifyKeyboardShortcuts();
-
+   
+   // Terminal
+   public abstract AppCommand newTerminal();
+   public abstract AppCommand renameTerminal();
```

Comment:
You don't need all of the `AppCommand` infrastructure to add buttons to the toolbar -- just make a menu item with a callback. `AppCommands` are global, can only have a single listener, and are exposed to the user (in the form of rebindable keyboard shortcuts) so they are not often used for contextual toolbar commands.